Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -1381,26 +1381,116 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
+                });
+
+            }
+            else{
+                this.title.appendChild(this._nodeTitleInfo().titleDOM);
+                this._createBadges();
+            }
             this._highlightResult = undefined;
         }
-        this._createBadges();
-
-        // Setting this.title will implicitly remove all children. Clear the
-        // selection element so that we properly recreate it if necessary.
         this._selectionElement = null;
         this.updateSelectionArea();
         this._highlightSearchResults();
         this._updatePseudoClassIndicator();
         this._updateBreakpointStatus();
     }
 
+    _buildAccessibilityDOM(accessibilityNode, baseURL)
+    {
+        var domNode = accessibilityNode.domNode;
+        var titleDOM = document.createDocumentFragment();
+        var tagElement = document.createElement("span");
+        tagElement.className = "accessibility-tree-node";
+        if (accessibilityNode.isIgnored) 
+            tagElement.classList.add("disabled-element");
+        
+        var tagNameElement = document.createElement("span");
+        let roleSpan = document.createElement("span");
+        roleSpan.className = "accessibility-role";
+        roleSpan.textContent = accessibilityNode.computedRole;
+        tagNameElement.appendChild(roleSpan);
+        if (accessibilityNode.computedLabel) {
+            let colon = document.createTextNode(": ");
+            let labelSpan = document.createElement("span");
+            labelSpan.className = "accessibility-label";
+            labelSpan.textContent = `"${accessibilityNode.computedLabel}"`;
+            tagNameElement.appendChild(colon);
+            tagNameElement.appendChild(labelSpan);
+        }
+        if (typeof accessibilityNode.focused === "boolean") {
+            let focusSpan = document.createElement("span");
+            focusSpan.className = "accessibility-focus";
+            focusSpan.textContent = ` Focusable: ${accessibilityNode.focused}`;
+            tagNameElement.appendChild(focusSpan);
+        }
+        var attributes = domNode.attributes();
+        for (var i = 0; i < attributes.length; ++i) {
+            var attr = attributes[i];
+            let rewrittenURL = absoluteURL(attr.value, baseURL);
+            if (rewrittenURL &&  (attr.name === "src" || /\bhref\b/.test(attr.name))) {
+                let urlSpan = document.createElement("span");
+                urlSpan.className = "accessibility-base-url";
+                urlSpan.textContent = ` Url: ${rewrittenURL}`;
+                tagNameElement.appendChild(urlSpan);
+            }
+        }
+        this._addAccessibilityWarnings(tagNameElement, accessibilityNode);
+        tagElement.appendChild(tagNameElement);
+        titleDOM.appendChild(tagElement);
+        titleDOM.append("\u200B");
+        return titleDOM;
+    }
+
+    _addAccessibilityWarnings(tagNameElement, accessibilityNode)
+    {
+        if (accessibilityNode.hasWarnings) {
```

Comment:
The check for the existence of warnings should happen at the call site and skip calling this method if not necessary.